File: src/leetcodescala/LongestContinuousIncreasingSubsequence674.scala
package src.leetcodescala

object LongestContinuousIncreasingSubsequence674 {
  def findLengthOfLCIS(nums: Array[Int]): Int = {
    var maxLen = 0
    var curLen = 1

    if (nums == null || nums.length == 0)
      return maxLen

    for (i <- 1 until nums.length) {
      if (nums(i) <= nums(i-1)) {
        maxLen = math.max(curLen, maxLen)
        curLen = 1
      }
      else {
        curLen += 1
      }
    }
    return math.max(maxLen, curLen)
  }
}

File: src/leetcode/LongestContinuousIncreasingSubsequence674.java
package src.leetcode;


public class LongestContinuousIncreasingSubsequence674 {
    public int findLengthOfLCIS(int[] nums) {
        int maxLen = 0;
        int curLen = 1;
        if (nums == null || nums.length == 0)
            return maxLen;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] <= nums[i-1]) {
                maxLen = Math.max(maxLen, curLen);
                curLen = 1;
            }
            else {
                curLen++;
            }
        }
        return Math.max(maxLen, curLen);
    }
}
